### PR TITLE
Update to Github Actions

### PR DESCRIPTION
**Pull request by zachluna**
Migrate from Travis to Github Actions.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: CI
+
+on: [push, pull_request]
```
Comment (bshaffer) [inline .github/workflows/ci.yaml:3]: I suggest restricting the push to the master branch only

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+    run:
+      runs-on: ${{ matrix.operating-system }}
+      strategy:
+        matrix:
+          operating-system: [ubuntu-latest, windows-latest, macos-latest]
+          php-versions: ['7.1', '7.2', '7.3', '7.4']
```
Comment (bshaffer) [inline .github/workflows/ci.yaml:11]: We can get rid of 7.1 as we do not have a runtime for 7.1 and don't need to support it

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            uses: actions/checkout@v2
+
+          - name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+              php-version: ${{ matrix.php-versions }}
+              coverage: none
+  
+          - name: Composer update
+            run: composer self-update
```
Comment (bshaffer) [inline .github/workflows/ci.yaml]: I do not think this step is necessary. We can just rely on the built in composer version to be sufficient.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (zachluna) [reply .github/workflows/ci.yaml]: Changed push/pull restricted to master

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (zachluna) [reply .github/workflows/ci.yaml]: Done, thanks

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  push:
+    branches:
+      - master
+jobs:
+    run:
+      runs-on: ${{ matrix.operating-system }}
+      strategy:
+        matrix:
+          operating-system: [ubuntu-latest, windows-latest, macos-latest]
+          php-versions: [7.2', '7.3', '7.4']
```
Comment (bshaffer) [inline .github/workflows/ci.yaml:16]: typo here - you're missing a `'`

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          run: composer install --prefer-dist --no-interaction
+
+        - name: Create log directory
+          run: mkdir -p build/logs
+
+        - name: Run phpunit
+          run: phpunit --coverage-clover build/logs/clover.xml
+
+        - name: Coveralls
+          run: vendor/bin/coveralls -v
```
Comment (bshaffer) [inline .github/workflows/ci.yaml:42]: nit: add newline

Comment (zachluna) [reply .github/workflows/ci.yaml:42]: Done, added new line at bottom of file.

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (zachluna) [reply .github/workflows/ci.yaml:16]: Done, thanks!

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+          uses: actions/checkout@v2
+          uses: php-actions/composer@v5
+          uses: php-actions/phpunit@v2
+
+        - name: Setup PHP
+          uses: shivammathur/setup-php@v2
+          with:
+            php-version: ${{ matrix.php-versions }}
+            coverage: xdebug
+            tools: php-cs-fixer, phpunit
```
Comment (bshaffer) [inline .github/workflows/ci.yaml:28]:
```suggestion
            tools: php-cs-fixer
```

**Review by bshaffer — COMMENTED** [review]
also, run `composer require --dev phpunit/phpunit:^8` and commit the changes it makes to `composer.json`

File: .github/workflows/ci.yaml (hunk 6 above)
Comment (zachluna) [reply .github/workflows/ci.yaml]: Done, thanks

File: .github/workflows/ci.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+        - name: Install dependencies     
+          run: composer install --prefer-dist --no-interaction
+
+        - name: Create log directory
+          run: mkdir -p build/logs
+
+        - name: Run phpunit
+          run: phpunit --coverage-clover build/logs/clover.xml
```
[the remark this answers is not in the document]
Comment (zachluna) [reply .github/workflows/ci.yaml]: Done, thanks

**Review by bshaffer — APPROVED** [review]
LGTM as long as we make sure to remove TODOs before we release!

**Pull request merged**